pkg/kv/kvserver/replica_app_batch.go

```diff
@@ -177,9 +177,8 @@ func (b *replicaAppBatch) Stage(
 	return cmd, nil
 }
 
-// stageWriteBatch applies the command's write batch to the application batch's
-// RocksDB batch. This batch is committed to Pebble in replicaAppBatch.commit.
-func (b *replicaAppBatch) stageWriteBatch(ctx context.Context, cmd *replicatedCmd) error {
+// addWriteBatch adds the command's writes to the batch.
+func (b *replicaAppBatch) addWriteBatch(ctx context.Context, cmd *replicatedCmd) error {
```

Comment:
The surrounding method is called `Stage` (and this is a reasonable name), so I would like a different word for the "inner" stage to avoid confusion. Are you sure "addWriteBatch` is too generic? This literally describes what this does, which is add the write batch (to the replicaAppBatch). I also don't want to name this `applyWriteBatch` since we're not actually sending it to the storage engine yet.

Let's discuss this on synchronously.

Reply:
Understood. Yeah, there is a trade-off between having unique names for precision, and having fewer entities in the mental model (stage vs stage&add).

My mental model was that there are 2 big steps: stage and apply. Then we can break down each into pre-thing, the-thing, and post-thing. So pre-stage and post-stage would be still associated with and part of "stage", just intercepting at the front or at the back of it (before and after the main job is done).

Reply:
"apply" to me is the act of committing the `pebble.Batch`. That is when the entries really become visible. Don't forget, this is batched! You basically have

- make pebble batch
- stage entry 1
  - pre-add trigger
  - add to pebble batch
  - post-add trigger
- stage entry 2
  - pre-add trigger
  - add to pebble batch
  - post-add trigger
- apply (commit the batch)

Was this reflected in your comment above?

Reply:
Yes, same to me. So my model was something like:

- make pebble batch
- for each entry call `Stage` which does
  - pre-stage
  - stage (into pebble batch)
  - post-stage
- pre-apply (non-existent, because post-stage probably does the same job)
- apply (write the batch)
- for each entry
  - post-apply

Reply:
We discussed, new idea:

use stage inside of c.Stage
rename c.Stage to c.Add (TODO)